RocketChat/Rocket.Chat.ReactNative
# [FIX] Toggle password visibility button not working

## Proposed changes

The "reveal password" button does not seem to be working and is expected that the password should be shown when the icon is tapped, and hidden when tapped again.

So the BordelessButton was switched to Touchable because the borderless button didn't appear above the text input's view for Android, so the click never worked. But worked fine for iOS. Although now with the Touchable, the click event is occurring in all views or modals there is the "reveal password" button. 

## Issue(s)	


## How to test or reproduce


The locales that appear the "reveal password" button are: 
- the Register view to create an account
- Login,
- E2EE reset password and on modal when you click on “reset E2E key” and pops up a modal with enter your password,
-  Profile View to set a New Password,
- enter Two Factor,
-  Join Code to enter a public room with code (you can test with channel #joincodetest1234, with the code 123456)
- The Room to edit info of the channels and change the Join Password

## Screenshots

## Types of changes




## Checklist



## Further comments
